uxarray/grid.py

```diff
@@ -371,3 +372,111 @@ def integrate(self, var_ds, quadrature_rule="triangular", order=4):
         integral = np.dot(face_areas, face_vals)
 
         return integral
+
+    def _populate_cartesian_xyz_coord(self):
+        """A helper function that populates the xyz attribute in UXarray.ds.
+        This function is called when we need to use the cartesian coordinates
+        for each node to do the calculation but the input data only has the
+        "Mesh2_node_x" and "Mesh2_node_y" in degree.
+
+        Note
+        ----
+        In the UXarray, we abide the UGRID convention and make sure the following attributes will always have its
+        corresponding units as stated below:
+
+        Mesh2_node_x
+         unit:  "degree_east" for longitude
+        Mesh2_node_y
+         unit:  "degrees_north" for latitude
+        Mesh2_node_z
+         unit:  "m"
+        Mesh2_node_cart_x
+         unit:  "m"
+        Mesh2_node_cart_y
+         unit:  "m"
+        Mesh2_node_cart_z
+         unit:  "m"
+        """
+
+        # Check if the cartesian coordinates are already populated
+        if "Mesh2_node_cart_x" in self.ds.keys():
+            return
+
+        # check for units and create Mesh2_node_cart_x/y/z set to self.ds
+        num_nodes = self.ds.Mesh2_node_x.size
+        node_cart_list_x = [0.0] * num_nodes
+        node_cart_list_y = [0.0] * num_nodes
+        node_cart_list_z = [0.0] * num_nodes
+        for i in range(num_nodes):
+            if "degree" in self.ds.Mesh2_node_x.units:
+                node = [
+                    np.deg2rad(self.ds["Mesh2_node_x"].values[i]),
+                    np.deg2rad(self.ds["Mesh2_node_y"].values[i])
+                ]  # [lon, lat]
+                node_cart = _convert_node_lonlat_rad_to_xyz(node)  # [x, y, z]
```

Comment:
Same idea as last comment, passing through the entire `lon` and `lat` arrays would return our nodes in carteisian coordinates with shape $(3, N_{nodes})$